### DISCUSSION TITLE
Blaupunkt Orlando cm-127

### DISCUSSION THREAD

**Discussion by Jamesdmahala** [tomaskovacik/vwcdavr#17] [General]
I have a Blaupunkt radio I picked up and have been using but am interested in getting it to have Bluetooth, and hope I could use the controls on the unit to control the Bluetooth module, and control my phone such as next and last track. I was going to use your code in hopes that the protocol was the same however the connections for the data in and out are a single pin how could I change your code to possibly get it to work with my head unit assuming it used the same protocol and if not how would I figure out the protocol of my unit. I am new to using the Arduino and coding however I do have a Arduino Uno r3, I am also curious as to what Bluetooth module you would suggest I use for this project. My brother codes for a living and I do have him at my disposal if I need help modifying code. I will link manuals for a compatible cd changer, the user manual for the unit it self and the schematic. I am a can solder and have access to components and pcb fabrication if I would need to make my own board or something. Thanks for any help.

James

[blaupunkt_orlando_cm127 (1).pdf](https://github.com/tomaskovacik/vwcdavr/files/7634745/blaupunkt_orlando_cm127.1.pdf)
[orlando_cm127.pdf](https://github.com/tomaskovacik/vwcdavr/files/7634747/orlando_cm127.pdf)
[cdc_a071.pdf](https://github.com/tomaskovacik/vwcdavr/files/7634748/cdc_a071.pdf)

**Comment by tomaskovacik** [comment]
hello,
I do not have time to investigate this, but in your place, I would look on the internet, there is a lot of information about this topic:
https://www.google.com/search?channel=fs&q=blaupunk+CDC+protocol

sorry
t.

**Reply by Jamesdmahala** [reply]
Thanks for your response I will try and get it working, do you have a suggestion for a Bluetooth module that will allow me to use all the buttons on my headunit to control my phone and can be controlled via my arduino and hopefully bluetooth 5.0

**Comment by tomaskovacik** [comment]
use BK3266 module, this one https://www.aliexpress.com/item/1005001682230861.html. go for "EnTone Nokey" version. It use same ascii protocol as bk3254, so you can take inspiration from my library: https://github.com/tomaskovacik/BK3254
